src/hardware/hantek-dso/protocol.c

```diff
@@ -241,7 +241,187 @@ static int get_channel_offsets(const struct sr_dev_inst *sdi)
 	return SR_OK;
 }
 
-static int dso_set_trigger_samplerate(const struct sr_dev_inst *sdi)
+
+static void dso2250_set_triggerpos(int value, int long_buffer, uint8_t dest[], int offset)
+{
+	uint32_t min, max;
+	uint32_t tmp, diff;
+
+	min = long_buffer ? 0 : 0x7d7ff;
+	max = 0x7ffff;
+
+	diff = max - min;
+	tmp = min + diff * value / 100;
+	sr_dbg("2250 trigger pos: %3d%% * [0x%x,0x%x] == 0x%x", value, min, max, tmp);
+
+	dest[offset + 0] = tmp & 0xff;
+	dest[offset + 1] = (tmp >> 8) & 0xff;
+	dest[offset + 2] = (tmp >> 16) & 0x7;
+}
+
+
+/* See http://openhantek.sourceforge.net/doc/namespaceHantek.html#ac1cd181814cf3da74771c29800b39028 */
+static int dso2250_set_trigger_samplerate(const struct sr_dev_inst *sdi)
+{
+	struct dev_context *devc;
+	struct sr_usb_dev_inst *usb;
+	int ret, tmp;
+	int base;
+	uint8_t cmdstring[12];
+	int trig;
+
+
+	sr_dbg("Preparing CMD_2250_SET_TRIGGERSOURCE.");
+
+	devc = sdi->priv;
+	usb = sdi->conn;
+
+	memset(cmdstring, 0, sizeof(cmdstring));
+	/* Command */
+	cmdstring[0] = CMD_2250_SET_TRIGGERSOURCE;
+	sr_dbg("Trigger source %s.", devc->triggersource);
+	if (!strcmp("CH2", devc->triggersource))
+		tmp = 3;
+	else if (!strcmp("CH1", devc->triggersource))
+		tmp = 2;
+	else if (!strcmp("EXT", devc->triggersource))
+		tmp = 0;
+	else {
+		sr_err("Invalid trigger source: '%s'.", devc->triggersource);
+		return SR_ERR_ARG;
+	}
+	cmdstring[2] = tmp;
+
+
+	sr_dbg("Trigger slope: %d.", devc->triggerslope);
+	cmdstring[2] |= (devc->triggerslope == SLOPE_NEGATIVE ? 1 : 0) << 3;
+
+	if (send_begin(sdi) != SR_OK)
+		return SR_ERR;
+
+	if ((ret = libusb_bulk_transfer(usb->devhdl, DSO_EP_OUT,
+			cmdstring, 8, &tmp, 100)) != 0) {
+		sr_err("Failed to set trigger/samplerate: %s.",
+		       libusb_error_name(ret));
+		return SR_ERR;
+	}
+	sr_dbg("Sent CMD_2250_SET_TRIGGERSOURCE.");
+
+
+	/* Frame size */
+	sr_dbg("Frame size: %d.", devc->framesize);
+	cmdstring[0] = CMD_2250_SET_RECORD_LENGTH;
+	cmdstring[2] = devc->framesize == FRAMESIZE_SMALL ? 0x01 : 0x02;
+
+	if (send_begin(sdi) != SR_OK)
+		return SR_ERR;
+
+	if ((ret = libusb_bulk_transfer(usb->devhdl, DSO_EP_OUT,
+			cmdstring, 4, &tmp, 100)) != 0) {
+		sr_err("Failed to set record length: %s.",
+		       libusb_error_name(ret));
+		return SR_ERR;
+	}
+	sr_dbg("Sent CMD_2250_SET_RECORD_LENGTH.");
+
+
+	memset(cmdstring, 0, sizeof(cmdstring));
+	cmdstring[0] = CMD_2250_SET_SAMPLERATE;
+	sr_dbg("Sample rate: %u", devc->samplerate);
+	base = 100e6;
+	if (devc->samplerate > base) {
+		/* Timebase fast */
+		sr_err("Sample rate > 100MHz not yet supported.");
+		return SR_ERR_ARG;
+
+		if (devc->framesize != FRAMESIZE_SMALL) {
```

Comment:
So, all of this is dead code, with no indication of whether it works, or is meant to work, or is just garbage.  Just delete it.